<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWin</class>
 <widget class="QMainWindow" name="MainWin">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>478</width>
    <height>348</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Microscope</string>
  </property>
  <widget class="QWidget" name="centralwidget"/>
  <widget class="QMenuBar" name="mbMain">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>478</width>
     <height>18</height>
    </rect>
   </property>
   <widget class="QMenu" name="mFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <widget class="QMenu" name="mRecent">
     <property name="title">
      <string>Recent images</string>
     </property>
    </widget>
    <addaction name="actReset"/>
    <addaction name="separator"/>
    <addaction name="actAppendImage"/>
    <addaction name="mRecent"/>
    <addaction name="separator"/>
    <addaction name="actSaveImage"/>
    <addaction name="actSaveAllImages"/>
    <addaction name="separator"/>
    <addaction name="actExit"/>
   </widget>
   <widget class="QMenu" name="menu_Help">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="actAbout"/>
   </widget>
   <widget class="QMenu" name="menu_Hardware">
    <property name="title">
     <string>&amp;Hardware</string>
    </property>
    <addaction name="actConnDevices"/>
    <addaction name="actDisconnDevices"/>
    <addaction name="separator"/>
    <addaction name="actConnController"/>
    <addaction name="actConnCamera"/>
   </widget>
   <addaction name="mFile"/>
   <addaction name="menu_Hardware"/>
   <addaction name="menu_Help"/>
  </widget>
  <widget class="QStatusBar" name="sbMain"/>
  <widget class="QToolBar" name="tbNormal">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actAppendImage"/>
   <addaction name="actDeleteImage"/>
   <addaction name="actImageStitching"/>
  </widget>
  <widget class="QToolBar" name="tbCamera">
   <property name="windowTitle">
    <string>toolBar_2</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actTakeImage"/>
   <addaction name="actAutoCamStit"/>
  </widget>
  <action name="actTakeImage">
   <property name="icon">
    <iconset resource="../rsrc/mainresources.qrc">
     <normaloff>:/icons/icons/edit-download.svg</normaloff>:/icons/icons/edit-download.svg</iconset>
   </property>
   <property name="text">
    <string>Take Image</string>
   </property>
   <property name="toolTip">
    <string>Take image from live camera</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+T</string>
   </property>
  </action>
  <action name="actImageStitching">
   <property name="icon">
    <iconset resource="../rsrc/mainresources.qrc">
     <normaloff>:/icons/icons/link.svg</normaloff>:/icons/icons/link.svg</iconset>
   </property>
   <property name="text">
    <string>Stitch</string>
   </property>
   <property name="toolTip">
    <string>Start stiching images on board</string>
   </property>
  </action>
  <action name="actAbortCamStit">
   <property name="text">
    <string>Stop</string>
   </property>
   <property name="toolTip">
    <string>Abort hte camera stitching process</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actAbortAutoCamStit">
   <property name="text">
    <string>Stop</string>
   </property>
   <property name="toolTip">
    <string>Abort the auto camera stitch process</string>
   </property>
  </action>
  <action name="actAutoCamStit">
   <property name="icon">
    <iconset resource="../rsrc/mainresources.qrc">
     <normaloff>:/icons/icons/edit-select-all.svg</normaloff>:/icons/icons/edit-select-all.svg</iconset>
   </property>
   <property name="text">
    <string>Auto Camera Stitching</string>
   </property>
   <property name="toolTip">
    <string>Run the automatic camera stitching mode</string>
   </property>
  </action>
  <action name="actAbout">
   <property name="text">
    <string>About Microscope</string>
   </property>
   <property name="toolTip">
    <string>Microscope information</string>
   </property>
  </action>
  <action name="actConnController">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Connect controller</string>
   </property>
  </action>
  <action name="actConnCamera">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Connect camera</string>
   </property>
  </action>
  <action name="actConnDevices">
   <property name="text">
    <string>Connect all devices</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actDisconnDevices">
   <property name="text">
    <string>Disconnect all devices</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
  <action name="actSaveImage">
   <property name="icon">
    <iconset resource="../rsrc/mainresources.qrc">
     <normaloff>:/icons/icons/document-save.svg</normaloff>:/icons/icons/document-save.svg</iconset>
   </property>
   <property name="text">
    <string>Save stitched image</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actSaveAllImages">
   <property name="text">
    <string>Save all images</string>
   </property>
   <property name="toolTip">
    <string>Save the calculation results for particle distribution</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actExit">
   <property name="icon">
    <iconset resource="../rsrc/mainresources.qrc">
     <normaloff>:/icons/icons/application-exit.svg</normaloff>:/icons/icons/application-exit.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actReset">
   <property name="icon">
    <iconset resource="../rsrc/mainresources.qrc">
     <normaloff>:/icons/icons/edit-clear-history.svg</normaloff>:/icons/icons/edit-clear-history.svg</iconset>
   </property>
   <property name="text">
    <string>Reset all</string>
   </property>
   <property name="toolTip">
    <string>Reset the whole application</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actAppendImage">
   <property name="icon">
    <iconset resource="../rsrc/mainresources.qrc">
     <normaloff>:/icons/icons/insert-image.svg</normaloff>:/icons/icons/insert-image.svg</iconset>
   </property>
   <property name="text">
    <string>Append image</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actSwitchCameraView">
   <property name="text">
    <string>Switch Camera View</string>
   </property>
  </action>
  <action name="actDeleteImage">
   <property name="icon">
    <iconset resource="../rsrc/mainresources.qrc">
     <normaloff>:/icons/icons/edit-delete.svg</normaloff>:/icons/icons/edit-delete.svg</iconset>
   </property>
   <property name="text">
    <string>Delete</string>
   </property>
   <property name="toolTip">
    <string>Delete selected image from previews</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="../rsrc/mainresources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actAbortCamStit</sender>
   <signal>triggered()</signal>
   <receiver>MainWin</receiver>
   <slot>abortCameraStitching()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>242</x>
     <y>195</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actAbortAutoCamStit</sender>
   <signal>triggered()</signal>
   <receiver>MainWin</receiver>
   <slot>abortAutoCameraStitching()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>242</x>
     <y>195</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actAutoCamStit</sender>
   <signal>triggered()</signal>
   <receiver>MainWin</receiver>
   <slot>runAutoCameraStitching()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>361</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actAbout</sender>
   <signal>triggered()</signal>
   <receiver>MainWin</receiver>
   <slot>about()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>361</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actConnCamera</sender>
   <signal>triggered()</signal>
   <receiver>MainWin</receiver>
   <slot>connectCamera()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>361</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actConnController</sender>
   <signal>triggered()</signal>
   <receiver>MainWin</receiver>
   <slot>connectController()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>361</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actConnDevices</sender>
   <signal>triggered()</signal>
   <receiver>MainWin</receiver>
   <slot>connectDevices()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>361</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actDisconnDevices</sender>
   <signal>triggered()</signal>
   <receiver>MainWin</receiver>
   <slot>disconnectDevices()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>361</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actReset</sender>
   <signal>triggered()</signal>
   <receiver>MainWin</receiver>
   <slot>reset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>361</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actSaveImage</sender>
   <signal>triggered()</signal>
   <receiver>MainWin</receiver>
   <slot>saveImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>361</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWin</receiver>
   <slot>exit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>361</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actAppendImage</sender>
   <signal>triggered()</signal>
   <receiver>MainWin</receiver>
   <slot>openImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>361</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actTakeImage</sender>
   <signal>triggered()</signal>
   <receiver>MainWin</receiver>
   <slot>takeImageFromCamera()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>461</x>
     <y>351</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actImageStitching</sender>
   <signal>triggered()</signal>
   <receiver>MainWin</receiver>
   <slot>stitchImages()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>238</x>
     <y>173</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actSaveAllImages</sender>
   <signal>triggered()</signal>
   <receiver>MainWin</receiver>
   <slot>saveAllImages()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>238</x>
     <y>173</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actDeleteImage</sender>
   <signal>triggered()</signal>
   <receiver>MainWin</receiver>
   <slot>deleteImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>238</x>
     <y>173</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>stitchImages()</slot>
  <slot>abortCameraStitching()</slot>
  <slot>takeImageFromCamera()</slot>
  <slot>moveTest()</slot>
  <slot>abortAutoCameraStitching()</slot>
  <slot>runAutoCameraStitching()</slot>
  <slot>about()</slot>
  <slot>connectCamera()</slot>
  <slot>connectController()</slot>
  <slot>connectDevices()</slot>
  <slot>disconnectDevices()</slot>
  <slot>saveImage()</slot>
  <slot>saveResults()</slot>
  <slot>reset()</slot>
  <slot>exit()</slot>
  <slot>openImage()</slot>
  <slot>clearRecentImages()</slot>
  <slot>saveAllImages()</slot>
  <slot>deleteImage()</slot>
 </slots>
</ui>
